R/UK.R

```diff
@@ -495,6 +495,61 @@ UK <- R6::R6Class("UK",
           release_date = self$release_date
         )
       return(clean_data)
+    },
+
+    #' @description Run tests on UK class.
+    #' @param download logical. To download the data (TRUE) or use a snapshot
+    #' (FALSE). Defaults to FALSE.
+    #' @importFrom testthat test_that expect_true expect_s3_class
+    #' @importFrom dplyr slice_tail
+    test = function(download = FALSE) {
+      super$test(download)
+      if (self$level == "1") {
+        data_name <- "UK level 1 with 'nhsregions=TRUE'"
+        self$nhsregions <- TRUE
```

Comment:
I think we only want to test the settings currently active in the class and not all the optional settings? This does make things a bit trickier though of course.

Reply:
What do you mean by active functions? Like only the settings for that class, such as if the level is "1" etc. ?

Reply:
yes - that its true of the validation use case where someone wants to know the integrity of their data. For the does this package work use case you want to test all possible option combinations of course. Hence the added complexity.

Reply:
If a user is testing a class interactively they would call `$test()` which runs the tests on that class. So if they have set it up like:

```
a <- UK$new(level = "1")
a$test(snapshot_path = "path/2/some/file.rds")
```

It would run the test on the level 1 data, e.g. `test_cleaning(), test_processing(), test_return(), specific_tests()`.

Running the package level tests with testthat calls each combination (e.g. UK level 1, UK level 2), sets up the class and then calls the test method on each one.

Will that do it? or is there something I'm missing?

Reply:
Mostly. I think passing snapshot for them probably doesn't make sense and as implemented this will test all possible options and not just the ones they are using. 

The other thing this is missing is the split between testing and validating. If my interest is in whether or not my data is robust then you only need a subset of tests and probably want to test your actual class without alterations rather than using a clone of the class. 

You could argue for tests as they are and another function that runs tests only on the data that is for analysis use (so  returned maybe). The use case is then a check after you have performed all the actions you are. going to with that class rather than the testing performing the actions. Is that somewhat clear?